### ISSUE TITLE
user rules path

### ISSUE THREAD

**Issue by black-body** [sqlfluff/sqlfluff#2352]
Hello everyone,
I am not sure is this enhancement or it is already there, if it is sorry for opening this issue.
My question is:
Is there a simple way to include user rules located at the user path using sqlfluff cli. 
g.e. 
 ```bash
echo "SELECT col1 FROM table1;" |  sqlfuff lint - --rulepath "/my/custom/path/rule_*.py"
```

**Comment by jpy-git** [comment]
So the project supports plugins for you to create your own rules: https://docs.sqlfluff.com/en/stable/developingplugins.html
and there's also an example of how you would go about doing that: https://github.com/sqlfluff/sqlfluff/tree/main/plugins/sqlfluff-plugin-example/src

Personally I think adding the `--rulepath` CLI option is more messy than using the already provided plugin method. If you use the plugin method you should be able to include/exclude rules (including your additional ones) via the usual config/CLI methods.

**Comment by black-body** [comment]
Yeah I was checking that as well, I was thinking is there a faster way :D as I am not too familiar with plugin development. Anyway I will probably proceed with plugin method for now, Thanks for quick response! I'll close this.

**Comment by jpy-git** [comment]
Echo @tunetheweb here, we could def use some improvement on our documentation in this area, so feel free to ping us here or on slack for clarifications and any suggestions/contributions for improving the docs based on your experience would be very helpful 😄

**Comment by black-body** [comment]
Sure thing, will do that! Thanks!

**Comment by black-body** [comment]
Hi,
I'll just start writing here on what issues I stumbled upon so far in setting up custom rules using plugin.

- Started from this [page](https://docs.sqlfluff.com/en/stable/developingplugins.html)
- I had to do some reading regarding plugins in python, [here](https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/)
- Did some reading on how to package python file, [here](https://packaging.python.org/en/latest/tutorials/packaging-projects/)

Hiccups during following example from linked in start page

- Hat to figure out importance of PLUGIN_LOGICAL_NAME and PLUGIN_ROOT_MODULE and their relationship with entry points in given example
- Plugin name restriction is no _ or -, hence use only CamelCase convention (This is implicitly mentioned in start page, it could be a little better presented at front)
- Class name convention is Rule_PluginName_LXXX and module containing the class is PluginName_LXXX.py (Again first is implicitly mentioned, and latter somehow but no explicit description on these)
- To confirm my rules are included I used 
```bash
sqlfluff rules
```
Noticed bad formating like this:
```bash
VpLint_L00: File should start with block comment                                
1 
```                                            
Actual name is VpLint_L001

Hiccups during writing custom rule:
- Didn't find any directions on this (Used standard rules to figure out the logic)

.... during testing:
- Followed the example but nowhere mentioned how to create test cases (how to specify dialect in test case)
- All test marked with pass_str are passing no matter what
- All test marked with fail_str fail no matter what
(I am still debuging above ones)

So far this is briefly some issues I stumbled upon, please forgive me if there are documentation on any items mentioned above I didn't spent too much time searching it as I tried to figure out everything from source code. Even if there is somewhere it would be good that is linked to that start page.
